tools/github_update.py

```diff
@@ -172,20 +172,49 @@ def set_commit_status(self, context_label, state, message, details_url):
 class GithubStatusUpdater(object):
 
     def __init__(self, default_context_label=''):
-        info = RepoInfo()
-        self._api = GithubAPI(info.repo_orgname(), info.commit_sha(), info.github_auth_token())
+        if self._should_access_github():
+            info = RepoInfo()
+            self._api = GithubAPI(info.repo_orgname(), info.commit_sha(), info.github_auth_token())
+        else:
+            self._api = None
         self._default_context_label = default_context_label
 
 
+    def _should_access_github(self):
+        '''returns whether the local environment appears to be running in CI'''
+        if 'WORKSPACE' not in os.environ:
+            # not running in CI. skip actually sending anything to GitHub
+            return False
+        if os.environ.get('GITHUB_DISABLE', ''):
+            # environment has notifications disabled. skip actually sending anything to GitHub
+            return False
+        if not (os.environ.get('GITHUB_COMMIT_STATUS_URL') or os.environ.get('BUILD_URL')):
+            # CI job was not triggered by a PR
+            return False
+        return True
+
+
     def list_contexts(self):
         '''returns a set of context labels of all statuses in a given commit'''
+        if not self._api:
+            return set([])
         return set([status['context'] for status in self._api.get_commit_statuses()])
 
 
     def update(self, state, message='', details_url='', context_label=''):
         '''sends a commit status update to github.
         returns True on success or False otherwise.
         state should be one of the values in 'VALID_STATES'.'''
+        if details_url:
+            logmsg = '{} {}: {} ({})'.format(context_label, state, message, details_url)
+        else:
+            logmsg = '{} {}: {}'.format(context_label, state, message)
```

Comment:
These logs are the reason we're bothering with this object handling the no-github at all?